presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -442,6 +444,11 @@ protected Type visitLogicalBinaryExpression(LogicalBinaryExpression node, Stacka
         @Override
         protected Type visitComparisonExpression(ComparisonExpression node, StackableAstVisitorContext<Context> context)
         {
+            Optional<ExpressionUtils.DynamicFilter> dynamicFilter = getDynamicFilterOptional(node);
```

Comment:
Because dynamic filter is always a comparison expressions. It compares some column reference with a deferred expression.